python-avd/pyavd/api/_anta/avd_fabric_data.py

```diff
@@ -106,20 +107,19 @@ def _get_vtep_ips(structured_config: dict[str, Any]) -> tuple[IPv4Address | None
 
         # Get the VTEP IP
         vxlan_source_interface = get(structured_config, "vxlan_interface.vxlan1.vxlan.source_interface")
+        loopback_interfaces = get(structured_config, "loopback_interfaces", default=[])
+        dps_interfaces = get(structured_config, "dps_interfaces", default=[])
         if vxlan_source_interface is not None:
-            # Determine which interface model to search based on source interface type
-            if "Dps" in vxlan_source_interface:
-                interface_model = get(structured_config, "dps_interfaces", default=[])
-            else:
-                interface_model = get(structured_config, "loopback_interfaces", default=[])
+            # Determine which interface model to pick based on source interface type
+            interface_model = dps_interfaces if "Dps" in vxlan_source_interface else loopback_interfaces
 
             vtep_ip_str = get(get_item(interface_model, "name", vxlan_source_interface, default={}), "ip_address")
             vtep_ip = vtep_ip_addr if vtep_ip_str and isinstance((vtep_ip_addr := ip_interface(vtep_ip_str).ip), IPv4Address) else None
 
         # Get the MLAG VTEP IP (for Multi-VTEP MLAG feature)
         vxlan_mlag_source_interface = get(structured_config, "vxlan_interface.vxlan1.vxlan.mlag_source_interface")
-        if vxlan_mlag_source_interface is not None:
-            mlag_vtep_ip_str = get(get_item(interface_model, "name", vxlan_mlag_source_interface, default={}), "ip_address")
+        if vxlan_mlag_source_interface is not None and "Dps" not in vxlan_mlag_source_interface:
```

Comment:
discussed on call - we should one day enforce valid pattern in schema
```suggestion
        if vxlan_mlag_source_interface is not None:
```